frontend/src/components/NewRuleForm.vue

```diff
@@ -63,31 +69,44 @@ const handleTrackingRuleSelected = (name: string) => {
 const handleGenerationRulesChanged = (rules: GenerationRule[]) => {
   generationRulesCount.value = rules.length;
 };
+
+const onCancel = () => {
+  router.push("/");
+};
+
+// Warning on tab close or hard navigation away
+const beforeUnloadHandler = (e: BeforeUnloadEvent) => {
+  if (!isDirty.value) return;
+  e.preventDefault();
+  e.returnValue = "";
+};
+onMounted(() => {
+  window.addEventListener("beforeunload", beforeUnloadHandler);
+});
+onBeforeUnmount(() => {
+  window.removeEventListener("beforeunload", beforeUnloadHandler);
+});
+
+// Warn on in-app navigation away/route changes
+onBeforeRouteLeave((_to, _from, next) => {
+  if (!isDirty.value) {
+    next();
+    return;
+  }
+  if (window.confirm("Discard new rule?")) next();
```

Comment:
Oh, sorry, that's my fault for being unclear. I wasn't suggesting that the _entire_ message be just that question, I was merely suggesting to use it as the _prompt_ at the end of @abompard's suggested message. So, more like...

```suggestion
  if (window.confirm("You have not saved the new rule yet. Discard new rule?")) next();
```

or

```suggestion
  if (window.confirm("You have not saved the new rule yet. Discard changes?")) next();
```

Reply:
Message corrected @ferdnyc

Reply:
LGTM, but I'm nobody here, my review carries no weight (procedurally). @abompard or another committer will have to weigh in.

Reply:
@abompard Over to you. Let's go! 💪